pkg/utils/resources/resources.go

```diff
@@ -93,20 +95,58 @@ func Subtract(lhs, rhs v1.ResourceList) v1.ResourceList {
 	return result
 }
 
-// Ceiling calculates the max between the sum of container resources and max of initContainers
+// inspired from https://github.com/kubernetes/kubernetes/blob/e2afa175e4077d767745246662170acd86affeaf/pkg/api/v1/resource/helpers.go#L96
+// https://kubernetes.io/blog/2023/08/25/native-sidecar-containers/
+// Ceiling calculates the max between the sum of container resources and max of initContainers along with sidecar feature consideration
 func Ceiling(pod *v1.Pod) v1.ResourceRequirements {
 	var resources v1.ResourceRequirements
+
 	for _, container := range pod.Spec.Containers {
 		resources.Requests = MergeInto(resources.Requests, MergeResourceLimitsIntoRequests(container))
 		resources.Limits = MergeInto(resources.Limits, container.Resources.Limits)
 	}
+
+	restartableInitContainerReqs := v1.ResourceList{}
```

Comment:
Practically speaking, I really like what you originally did where you broke out the limits and resources logic. I thought that was super good 🎉 IMO I would go back to that since it simplifies things and is closer to upstream. I think we just want to make sure that we do the limits merge logic for the resource requests right now